packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,363 @@
+  // e.g. cron-job and system restart (depending on the os) we could omit
+  // this call `await this.fs.delete(this.backupFile)`.
+  private async cleanBackup() {
+    this.agent.config.logger?.trace(`Deleting the backup file at '${this.backupFile}'`)
+    await this.fs.delete(this.backupFile)
+  }
+
+  /**
+   * Move the migrated and updated database file to the new location according
+   * to the `FileSystem.dataPath`.
+   */
+  private async moveToNewLocation() {
+    const src = this.dbPath
+    // New path for the database
+    const dest = this.newWalletPath
+
+    // create the wallet directory
+    await this.fs.createDirectory(dest)
+
+    this.agent.config.logger?.trace(`Moving upgraded database from ${src} to ${dest}`)
+
+    // Copy the file from the database path to the new location
+    await this.fs.copyFile(src, dest)
+
+    // Delete the original, only if specified by the user
+    if (this.deleteOnFinish) await this.fs.delete(this.dbPath)
+  }
+
+  /**
+   * Function that updates the values from an indy-sdk structure to the new askar stucture.
+   *
+   * - Assert that the paths that will be used are free
+   * - Create a backup of the database
+   * - Update the Keys
+   * - Update the Master Secret (Link Sercet)
+   * - Update the credentials
+   * If any of those failed:
+   *   - Revert the database
+   * - Clear the backup from the temporary directory
+   */
+  public async update() {
+    await this.assertDestinationsAreFree()
+
+    await this.backupDatabase()
+    try {
+      await this.updateKeys()
+      await this.updateMasterSecret()
+      await this.updateCredentials()
+
+      await this.moveToNewLocation()
+    } catch (cause) {
+      this.agent.config.logger?.error('Migration failed. Reverting state.')
+
+      await this.revertDatabase()
+
+      throw new IndySdkToAskarMigrationError('Migration failed. State has been reverted.', { cause })
+    } finally {
+      await this.cleanBackup()
+    }
+  }
+
+  private async updateKeys() {
+    const category = 'Indy::Key'
+
+    this.agent.config.logger?.trace(`[indy-sdk-to-askar-migration]: Updating category: ${category}`)
+
+    let updateCount = 0
+    const session = this.store.transaction()
+    for (;;) {
+      const txn = await session.open()
+      const keys = await txn.fetchAll({ category, limit: 50 })
+      if (!keys || keys.length === 0) {
+        await txn.close()
+        break
+      }
+
+      for (const row of keys) {
+        await txn.remove({ category, name: row.name })
+        const signKey: string = JSON.parse(row.value as string).signkey
+        const keySk = TypedArrayEncoder.fromBase58(signKey)
+        const key = Key.fromSecretBytes({
+          algorithm: KeyAlgs.Ed25519,
+          secretKey: keySk.subarray(0, 32),
+        })
+        await txn.insertKey({ name: row.name, key })
+        updateCount++
+      }
+      await txn.commit()
+    }
+
+    this.agent.config.logger?.trace(
```

Comment:
Can you reorder the severity a bit. I think start/finished for a category can be of level info, while each key that is migrated could be of severity debug. Even smaller steps could be trace etc...